src/libs/ascent/utils/ascent_string_utils.cpp

```diff
@@ -39,39 +50,243 @@ void split_string(const std::string &s,
+  if (dir_path.size() == 0) {
+    dir_path = ".";
   }
 
-  std::string result;
-  bool has_format = name.find("%") != std::string::npos;
-  if(has_format)
+  // Building a pattern to match to filenames
+  // These patterns will be used to identify locations with unknown numbers and 
+  std::regex family_pattern(R"(\{family:([a-zA-Z0-9.]*)\})");
+  std::regex other_fmts_pattern(R"(\{[a-zA-Z]*:[a-zA-Z0-9.]*\})");
+
+  // This patern is used to match to numbers.
+  // It is looking for integers, decimal, and scientific notation values
+  std::string number_pattern = R"([+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?)";
+  
+  std::string search_pattern_str = file_name_fmt;
+  std::smatch match;
+  if (std::regex_search(search_pattern_str, match, family_pattern))
   {
-    // allow for long file paths
-    char buffer[2048];
-    snprintf(buffer, 2048, name.c_str(), counter);
-    result = std::string(buffer);
+    // This adds a capturing group that matches to a decimal number
+    // When running a regex_search this value will be captured and saved out
+    search_pattern_str.replace(match.position(0), match.length(0), "(" + number_pattern + ")");
+  }
+  while (std::regex_search(search_pattern_str, match, other_fmts_pattern))
+  {
+    // This adds a the pattern for a decimal number
+    search_pattern_str.replace(match.position(0), match.length(0), number_pattern);
+  }
+
+  // If there are no formatting sections in the path, then the format should have been added to the
+  // end ot the path. This adds a capturing group that matches to a decimal number.
+  // When running a regex_search this value will be captured and saved out
+  if (search_pattern_str == file_name_fmt) {
+    search_pattern_str += "(" + number_pattern + ")";
+  }
+
+  // Use the defined pattern to make the final regular expression
+  // Adding a ^ to lock the pattern to the start of the file_name and a pattern for a file extension
+  // at the end to the pattern
+  search_pattern_str = "^" + search_pattern_str + R"(\.[a-zA-Z]+$)";
+  std::regex search_pattern(search_pattern_str);
+
+  // Checking the directory contents for any filenames that match
+  std::vector<std::string> contents;
+  conduit::utils::list_directory_contents(dir_path, contents);
+  for (const std::string& item : contents) {
+      std::string file_name, rm_path;
+      conduit::utils::rsplit_file_path(item, file_name, rm_path);
+      
+      std::smatch file_match;
+      if (std::regex_search(file_name, file_match, search_pattern)) {
+        int matched_value = static_cast<int>(std::stod(file_match[1].str()));
+        if (matched_value >= family_value)
+        {
+          family_value = matched_value + 1;
+        }
+      }
+  }
+
+  return family_value;
+}
+
+int get_family_value(const std::string& path_string, int family_value)
+{
+  
+  std::string modified_path_string = path_string;
+
+  int rank = 0;
+#ifdef ASCENT_MPI_ENABLED
+  MPI_Comm mpi_comm = MPI_Comm_f2c(flow::Workspace::default_mpi_comm());
```

Comment:
lets pass in `int mpi_comm_id`, instead of using comm form the workspace, that way we can use it generally